include/zephyr/dfu/mcuboot.h

```diff
@@ -255,6 +255,20 @@ int boot_request_upgrade(int permanent);
  */
 int boot_request_upgrade_multi(int image_index, int permanent);
```

Comment:
This function should be used, a new function taking a partition ID should not be introduced

Reply:
this one unfortunately does something different, that is for pairs.
it will only use the secondary slots. 
```
image_index = 0 will request on slot1_partition
image_index = 1 will request on slot3_partition
image_index = 2 will request on slot5_partition
```
and I can't request an update on slot0, which I need.

Reply:
look in `bootloader/mcuboot/boot/zephyr/include/sysflash/sysflash.h` and `/home/fmaass@vogl.zone/zephyr-apps/sw__otem_mcu/bootloader/mcuboot/boot/bootutil/src/bootutil_public.c`

Reply:
MCUboot operates in pairs so I don't understand the problem? Or are you saying you are using directXIP mode and it's marking the wrong slot? If so, then the function I pointed to should be updated to support XIP mode but adding a new function just taking a partition ID is not acceptable

Reply:
See functions in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

Reply:
It is marking the wrong slot. Its not just for directXIP, but also for ram load mode (even throw mcuboot currently don't hake use of the trailer, but could be extended the same way that DIRECT_XIP got extended to DIRECT_XIP_REVERT and use the same functionality)